File: backend/user_details/enums/diet_type.py
from enum import Enum


class DietType(str, Enum):
    FAT_LOSS = "fat_loss"
    MUSCLE_GAIN = "muscle_gain"
    WEIGHT_MAINTENANCE = "weight_maintenance"

    def get_diet_type_factor(self):
        return {
            DietType.FAT_LOSS: -1,
            DietType.MUSCLE_GAIN: 1,
            DietType.WEIGHT_MAINTENANCE: 0,
        }[self]

File: frontend/lib/models/user_details/diet_type.dart
enum DietType {
  fatLoss(0, 'FAT_LOSS'),
  muscleGain(1, 'MUSCLE_GAIN'),
  weightMaintenance(2, 'WEIGHT_MAINTENANCE'),
  vegetarian(3, 'VEGETARIAN'),
  vegan(4, 'VEGAN'),
  keto(5, 'KETO');

  final int value;
  final String nameStr;

  const DietType(this.value, this.nameStr);

  String toJson() => nameStr;

  static DietType fromJson(String value) {
    return DietType.values.firstWhere(
          (e) => e.nameStr == value.toUpperCase(),
      orElse: () => throw ArgumentError('Unknown diet type: $value'),
    );
  }

  int toInt() => value;

  static DietType fromInt(int value) {
    return DietType.values.firstWhere(
          (e) => e.value == value,
      orElse: () => throw ArgumentError('Invalid diet type value: $value'),
    );
  }
}
